graphistry/compute/predicates/is_in.py

```diff
@@ -1,21 +1,126 @@
-from typing import Any, List
+from typing import Any, List, Dict
 import pandas as pd
+import numpy as np
+from datetime import datetime, date, time
 
 from graphistry.utils.json import assert_json_serializable
 from .ASTPredicate import ASTPredicate
+from ..ast_temporal import TemporalValue, DateTimeValue, DateValue, TimeValue
+from ...models.gfql.coercions.temporal import to_native
+from ...models.gfql.types.guards import is_basic_scalar, is_any_temporal
+from ...models.gfql.types.predicates import IsInElementInput
 from graphistry.compute.typing import SeriesT
 
 
 class IsIn(ASTPredicate):
-    def __init__(self, options: List[Any]) -> None:
-        self.options = options
+    def __init__(self, options: List[IsInElementInput]) -> None:
+        self.options = self._normalize_options(options)
+    
+    def _normalize_options(self, options: List[IsInElementInput]) -> List[Any]:
+        """Normalize options list to handle temporal values
+        
+        Returns List[Any] because normalized values include pandas types
+        and other hashable types for membership testing.
+        """
+        normalized = []
+        for val in options:
+            normalized.append(self._normalize_value(val))
+        return normalized
+    
+    def _normalize_value(self, val: IsInElementInput) -> Any:
```

Comment:
this is confusing, should normalize to some named type made out of known variants